pkg/adapter/incoming.go

```diff
@@ -86,6 +121,97 @@
 	return true, repo, nil
 }
 
+func (l *listener) getAndUpdateInstallationID(ctx context.Context, req *http.Request, run *params.Run, repo *v1alpha1.Repository) error {
+	JWTtoken, err := GenerateJWT(ctx, run)
+	if err != nil {
+		return err
+	}
+
+	enterpriseURL := req.Header.Get("X-GitHub-Enterprise-Host")
+	if enterpriseURL != "" {
+		apiURL = enterpriseURL
+	}
+
+	res, err := getResponse(ctx, http.MethodGet, installationURL, JWTtoken, run)
+	if err != nil {
+		return err
+	}
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	installationData := []gt.Installation{}
+	if err = json.Unmarshal(data, &installationData); err != nil {
+		return err
+	}
+
+	/* each installationID can have list of repository
+	ref: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation ,
+	     https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-app-installation */
+	for i := range installationData {
+		res, err := getResponse(ctx, http.MethodPost, fmt.Sprintf(installationTokenURL, *installationData[i].ID), JWTtoken, run)
```

Comment:
the installation token is going to bound to every repository which could be a security issue since this would include private repos in a org. look at how it's done in parse_payload to make sure it's bounded to the repo where it's coming from unless the configuration setting is set (and please use go-github library for that)

Reply:
To get repository list used go-github library instead of API call